src/main/java/mate/academy/spring/service/impl/CustomUserDetailsService.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.spring.service.impl;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email).orElseThrow(
+                () -> new UsernameNotFoundException("Can't find user by email: " + email));
+        String[] roles = user.getRoles().stream()
+                .map(role -> role.getRoleType().name())
+                .toArray(String[]::new);
+        return org.springframework.security.core.userdetails.User
```

Comment:
move this to import

Reply:
It would be great if that were possible.